tests/msc2716_test.go

```diff
@@ -1164,42 +1164,6 @@ func includes(needle string, haystack []string) bool {
 	return false
 }
 
-func fetchUntilMessagesResponseHas(t *testing.T, c *client.CSAPI, roomID string, check func(gjson.Result) bool) {
```

Comment:
Moving this to `tests/room_timestamp_to_event_test.go` because we can't reference it from the `msc2716` build tag when it's not included.

We don't build with `msc2716` in the workers Complement environment (related https://github.com/matrix-org/synapse/pull/14074)